src/main/java/org/jenkinsci/plugins/buildnameupdater/BuildNameUpdater.java

```diff
@@ -0,0 +1,243 @@
+
+        listener.getLogger().println("Getting version from file: " + fp);
+
+        try {
+            version = fp.act(new MyFileCallable());
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            listener.getLogger().println("Macro evaluating failed with:");
+            e.printStackTrace();
+        }
+
+        listener.getLogger().println("Loaded version is " + version);
+        return version == null ? "" : version;
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl)super.getDescriptor();
+    }
+
+    public static class BuildNameFromFile{
+        private String buildName;
+
+        @DataBoundConstructor
+        public BuildNameFromFile(String buildName){
+            this.buildName = buildName;
+        }
+
+        @SuppressWarnings("unused")
+        public String getBuildName() {
+            return buildName;
+        }
+    }
+
+    public static class BuildNameFromMacro{
+        private String macroTemplate;
+        private boolean macroFirst;
+
+        @DataBoundConstructor
+        public BuildNameFromMacro(String macroTemplate, boolean macroFirst){
+            this.macroTemplate = macroTemplate;
+            this.macroFirst = macroFirst;
+        }
+
+        @SuppressWarnings("unused")
+        public String getMacroTemplate() {
+            return macroTemplate;
+        }
+
+        @SuppressWarnings("unused")
+        public boolean getMacroFirst() {
+            return macroFirst;
+        }
+    }
+
+    private static class MyFileCallable implements FilePath.FileCallable<String> {
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        public String invoke(File file, VirtualChannel channel) throws IOException, InterruptedException {
+            if (file.getAbsoluteFile().exists()){
+                BufferedReader br = new BufferedReader(new FileReader(file.getAbsoluteFile()));
+                return br.readLine();
+            } else {
+                return "";
+            }
+        }
+    }
+
+    @Extension // This indicates to Jenkins that this is an implementation of an extension point.
+    public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
+        public DescriptorImpl() {
+            load();
+        }
+
+        @SuppressWarnings("unused")
+        public FormValidation doCheckName(@QueryParameter String value)
+                throws IOException, ServletException {
+            if (value.length() == 0)
+                return FormValidation.error("Please set a file path");
+            return FormValidation.ok();
+        }
+
+        public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+            // Indicates that this builder can be used with all kinds of project types
+            return FreeStyleProject.class.isAssignableFrom(jobType);
```

Comment:
The comment is incorrect. You support Freestyle projects only.
BTW, I don't see any reliance in your code, so probably you could just return <code>true</code> here to support Matrix Projects and other common job types